src/palace/manager/integration/license/overdrive/api.py

```diff
@@ -589,6 +611,138 @@ def _get_book_list_page(
+        of book data. In this way, "page" retrievals are accelerated while allowing the client to retrieve chunks
+        in a deterministic and therefore retriable manner.
+        """
+        base_url = self.endpoint(self.HOST_ENDPOINT_BASE)
+        async with self.create_async_client(
+            connections=connections, base_url=base_url
+        ) as client:
+            urls: deque[str] = deque()
+            books: dict[str, Any] = {}
+            extractor_class = extractor_class or OverdriveRepresentationExtractor
+            urls.append(endpoint.url)
+            req = client.get(endpoint.url)
+            response = await req
+            data = response.json()
+            next_url = extractor_class.link(data, rel_to_follow)
+            next_endpoint: BookInfoEndpoint | None = (
+                BookInfoEndpoint(next_url) if next_url else None
+            )
+            async_task_list = list()
+            response_products = data["products"]
+            for product in response_products:
+                identifier = product["id"].lower()
+                books[identifier] = product
+                if fetch_metadata:
+                    async_task_list.append(
+                        self._get_metadata_async(base_url, product, client)
+                    )
+
+                if fetch_availability:
+                    async_task_list.append(
+                        self._get_availability_async(
+                            base_url,
+                            product,
+                            client,
+                        )
+                    )
+
+            await asyncio.gather(*async_task_list)
+
+            return list(books.values()), next_endpoint
+
+    async def _get_availability_async(
+        self, base_url: str, book_info: dict[str, Any], client: AsyncClient
+    ) -> None:
+        url = book_info["links"]["availabilityV2"]["href"].removeprefix(base_url)
+        data = await self._get_product_relation(client, url)
+        if data:
+            book_info["availabilityV2"] = data
+
+    async def _get_metadata_async(
+        self, base_url: str, book_info: dict[str, Any], client: AsyncClient
+    ) -> None:
+        url = book_info["links"]["metadata"]["href"].removeprefix(base_url)
+        data = await self._get_product_relation(client, url)
+        if data:
+            book_info["metadata"] = data
+
+    async def _get_product_relation(
+        self, client: AsyncClient, url: str
+    ) -> dict[str, Any] | None:
+        req = client.get(url)
+        response = await req
+        # We allow a 404 response code for availability or metadata since those links may not exist for a given
+        # identifier.
+        if response.status_code == 404:
+            return None
+        else:
+            data: dict[str, Any] = response.json()
+            return data
+
+    def create_async_client(
```

Comment:
Minor: Seems like this should be a private method, not sure we would ever expect this to be called from outside of the class. Maybe `_create_configured_async_client` or similar.